src/container/srv_container.c

```diff
@@ -2082,13 +2082,21 @@ cont_op_with_svc(struct ds_pool_hdl *pool_hdl, struct cont_svc *svc,
 		if (rc != 0)
 			D_GOTO(out_lock, rc);
 		rc = cont_op_with_cont(&tx, pool_hdl, cont, rpc);
-		cont_put(cont);
 	}
 	if (rc != 0)
 		D_GOTO(out_lock, rc);
 
 	rc = rdb_tx_commit(&tx);
+	if (rc != 0)
+		D_GOTO(out_lock, rc);
+
+	if (opc == CONT_SNAP_CREATE || opc == CONT_SNAP_DESTROY) {
+		D_ASSERT(cont != NULL);
+		ds_update_snap_iv(&tx, cont);
```

Comment:
Using tx after committing it is considered hacky; the expected way is to read the previous list of snapshots and add/remove the new/victim snapshot.

Reply:
What about starting another tx to read the snapshot list?

Reply:
Using another TX would be acceptable, though a bit less preferable than the expected way.

Reply:
ah, so update_snap_iv() is actually using the original snapshots?  Maybe you can fix this function?

Reply:
I discussed with Liwei, we agreed on that we can compromise to begin another tx to read and propagate snap list.

I updated the patch (the patch is squashed and force pushed), please review it again. Thanks.